src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentConfigCompare/DeploymentConfigCompare.tsx

```diff
@@ -0,0 +1,369 @@
+        placeholder: 'Select State',
+        classNamePrefix: 'environment-config-type-selector',
+        inputId: `environment-config-type-selector-${isCompare ? 'compare' : 'current'}`,
+        name: `environment-config-type-selector-${isCompare ? 'compare' : 'current'}`,
+        variant: SelectPickerVariantType.BORDERLESS,
+        isSearchable: false,
+        value: getOptionByValue(
+            getEnvironmentConfigTypeOptions(isCompare),
+            !isCompare
+                ? getPreviousDeploymentOptionValue(identifierId, pipelineId) || configType
+                : getPreviousDeploymentOptionValue(compareWithIdentifierId, compareWithPipelineId) ||
+                      compareWithConfigType,
+            {
+                label: BASE_CONFIGURATIONS.name,
+                value: '',
+            },
+        ),
+        onChange: onEnvironmentConfigTypeChange(isCompare),
+    })
+
+    const deploymentConfigDiffSelectors: DeploymentConfigDiffProps['selectorsConfig'] = {
+        primaryConfig: [
+            {
+                id: 'compare-with-environment-selector',
+                type: 'selectPicker',
+                selectPickerProps: renderCompareEnvironmentSelectorProps(),
+            },
+            ...(compareWithConfigType !== AppEnvDeploymentConfigType.DEFAULT_VERSION
+                ? [
+                      {
+                          id: `environment-config-type-selector-compare`,
+                          type: 'selectPicker' as const,
+                          selectPickerProps: renderEnvironmentConfigTypeSelectorProps(true),
+                      },
+                  ]
+                : []),
+        ],
+        secondaryConfig: [
+            {
+                id: envName || BASE_CONFIGURATIONS.name,
+                type: 'string',
+                text: envName || BASE_CONFIGURATIONS.name,
+            },
+            {
+                id: `environment-config-type-selector-current`,
+                type: 'selectPicker',
+                selectPickerProps: renderEnvironmentConfigTypeSelectorProps(),
+            },
+        ],
+    }
+
+    const getBackURL = () => {
+        const envId = getEnvironmentIdByEnvironmentName(environments, envName)
+        const basePath = pathname.split(`/${envName || URLS.APP_ENV_CONFIG_COMPARE}`)[0]
+        const additionalPath =
+            envId !== -1 ? `${URLS.APP_ENV_OVERRIDE_CONFIG}/${envId}/${resourceType}/${resourceName}` : resourceType
+
+        return `${generatePath(basePath, { appId })}/${additionalPath}`
+    }
+
+    // ERROR SCREEN
+    if (comparisonDataErr || optionsErr) {
+        return <ErrorScreenManager code={comparisonDataErr?.code || optionsErr?.code} />
+    }
+
+    return (
+        <DeploymentConfigDiff
+            isLoading={
+                (comparisonDataLoader || !appEnvDeploymentConfigList || optionsLoader) &&
+                !comparisonDataErr &&
+                !optionsErr
+            }
```

Comment:
In such scenario, when we click on the reload in error screen manager the loader is not shown. Pls modify the conditions if required